packages/vue-router/src/createApp.ts

```diff
@@ -97,8 +93,18 @@ export const createApp = (
             ? [
                 router.beforeEach((to, from) => {
                   if (from !== START_LOCATION) {
+                    // check if the next path (to.path) is inside the mfe (in its routes)
+                    const isToAppRoute = routes.find(
+                      ({ path }) =>
+                        path.replace(/\/$/, "") === to.path.replace(/\/$/, "")
+                    );
+
+                    const nextPathname = isToAppRoute
```

Comment:
I think we can remove this `nextPathname` ternary and do in the `onRemoteNavigate` underneath the following:
```
if (!isToAppRoute)
  onRemoteNavigate?.({
    pathname: to.path,
    hash: to.hash,
    // TODO search: to.query,
  });
```
This will be more performant because the host doesn't have to re-render or do any work since the navigation only affects this app